### ISSUE TITLE
Question on S3 support of LAPKC71

### ISSUE THREAD

**Issue by zh5j** [right-0903/ACPI_fix_for_KC57#1]
Hi, just came across this amazing project, thanks for sharing! 

One quick question, did you notice that the NUC kit does not resume properly from deep sleep (S3)? My LAPKC71E never resumes properly from S3, and some googling pointed to bunch of guys patching their DSDT tables to fix this. 

Thanks again for sharing!

**Comment by right-0903** [comment]
Yes and no, long long ago, when I was using Windows OS, someone told me S3 was killed to support S0/S2idle, but I remember I can S3 sleep with bios37 or so(judging by power meter, power less than 1w, S0 on windows far more than that). Later, I switched to Linux completely, once I dived into sleep, I tried to switch to deep, then I can't back anymore.
```
$ cat /sys/power/mem_sleep
[s2idle] deep
```
But I don't care about this, S2idle is ok for me on Linux. So I quit at that time.

Back to now, so, have you ever tested on Windows? 

I have a glance, I think issues with wake or sleep method in DSDT
```
...
    Method (_PTS, 1, NotSerialized)  // _PTS: Prepare To Sleep
    {
        If (Arg0)
        {
            \_SB.TPM.TPTS (Arg0)
            \_SB.PC00.NPTS (Arg0)
            RPTS (Arg0)
            \_SB.PC00.LPCB.SPTS (Arg0)
        }
    }

    Method (_WAK, 1, NotSerialized)  // _WAK: Wake
    {
        \_SB.PC00.LPCB.SWAK (Arg0)
        RWAK (Arg0)
        \_SB.PC00.NWAK (Arg0)
        If ((Arg0 == 0x04))
        {
            \_SB.PC00.LPCB.EC0.XSEC = \_SB.PC00.LPCB.EC0.OSEC
        }

        Return (AM00) /* \AM00 */
    }
...
```

1. `_PTS ` does not handle correctly, so that we can't resume
2. It is just `_WAK`'s fault.

If you can offer me one copy of DSDT from a TongFang machine(Mechrevo, XMG/Schenker, etc) which is shipped with Intel gen11 or so and whose S3 works well, then I may fix it.

**Comment by zh5j** [comment]
Thanks for the prompt reply! 

The annoying part is, although the function is broken, the BIOS still advertises support for S3:

```
(base) hl@hl-lapkc71e:~$ sudo dmesg|grep S3
[sudo] password for hl: 
[    0.237345] ACPI: PM: (supports S0 S3 S4 S5)

```
So nothing stops me from setting mem_sleep to `deep` and having the laptop freeze upon resume. Default behaviour out of the box for ubuntu2404. Annoying. 

For testing with Windows - I can try using Win10 LTSC2015/2019 which should be well before M$ enforcing modern standby, or later versions - which would you recommend?

For grabbing the DSDT from a working laptop - I'll try my best to source one of those but no promises here....

Thanks again for your time!

**Comment by zh5j** [comment]
Also if you could find the old BIOS you mentioned, could you please send me the file or mention a source that still hosts such files? The official site only hosts the latest 2 versions.

**Comment by right-0903** [comment]
> The annoying part is, although the function is broken, the BIOS still advertises support for S3:
> ```
> (base) hl@hl-lapkc71e:~$ sudo dmesg|grep S3
> [sudo] password for hl: 
> [    0.237345] ACPI: PM: (supports S0 S3 S4 S5)
> ```

The driver is written against ACPI, if specific sections presented, then they would be reported as supported.
```
    Name (_S0, Package (0x04)  // _S0_: S0 System State
    {
        Zero, 
        Zero, 
        Zero, 
        Zero
    })
    Name (_S3, Package (0x04)  // _S3_: S3 System State
    {
        0x05,
        Zero, 
        Zero, 
        Zero
    })
    Name (_S4, Package (0x04)  // _S4_: S4 System State
    {
        0x06,
        Zero, 
        Zero, 
        Zero
    })
    Name (_S5, Package (0x04)  // _S5_: S5 System State
    {
        0x07,
        Zero, 
        Zero, 
        Zero
    })
```

> For testing with Windows - I can try using Win10 LTSC2015/2019 which should be well before M$ enforcing modern standby, or later versions - which would you recommend?

No, I don't use it on bare metal anymore. I have no idea.
